core/src/main/java/org/infinispan/commands/functional/ReadWriteKeyCommand.java

```diff
@@ -130,4 +128,26 @@ public void init(ComponentRegistry componentRegistry) {
          ((InjectableComponent) f).inject(componentRegistry);
       }
    }
+
+   @Override
+   public final boolean isReturnValueExpected() {
+      return true;
+   }
+
+   @Override
+   public Metadata getMetadata() {
```

Comment:
I wouldn't mind had you provided an overload for `WriteEntryView.set()` with signature `Void set(V value, Metadata metadata)`, if you don't want to mess with conversions. But it's conceptually wrong to fetch data from a "special" `Function` instance, and apply them separately.